### DISCUSSION TITLE
Deleted Question

### DISCUSSION THREAD

**Discussion by WildBobby** [idaholab/moose#19151] [Q&A Modules: General] [answered]

**Comment by recuero** [comment, marked as answer]
Use of `ADHillElastoPlasticityStressUpdate` is experimental and this object is not fully optimized, so it could be slow.  If the material yields isotropically, then the best choice would probably be to use something like `IsotropicPlasticityStressUpdate` with the proper elasticity tensor. Some code changes may be required though.

**Reply by GiudGiud** [reply]
For a) , changing a source file from using a constant to using a function is pretty easy. So if you go that route you will have no problem doing that.

**Reply by GiudGiud** [reply]
I m suggesting you modify the source (code) file, not just the input file.

**Reply by recuero** [reply]
Using a hardening function for `ADHillElastoPlasticityStressUpdate` is not currently supported. It'd require a small _code_ change. Manipulating the input file isn't enough.

**Reply by GiudGiud** [reply]
also what is it a function of?
Unless it is ONLY space and time, you want to be making this coefficient a material property, not a Function object

**Reply by GiudGiud** [reply]
oh so no temperature dependence or any local variable, just plastic strain?

See the documentation for Function https://mooseframework.inl.gov/syntax/Functions/index.html
Material properties & materials https://mooseframework.inl.gov/syntax/Materials/index.html
